neurolib/models/multimodel/builder/thalamus.py

```diff
@@ -0,0 +1,442 @@
+
+from ..builder.base.constants import EXC, INH, LAMBDA_SPEED
+from ..builder.base.network import SingleCouplingExcitatoryInhibitoryNode
+from ..builder.base.neural_mass import NeuralMass
+
+DEFAULT_PARAMS_TCR = {
+    "tau": 20.0,  # ms
+    "Q_max": 400.0e-3,  # 1/ms
+    "theta": -58.5,  # mV
+    "sigma": 6.0,
+    "C1": 1.8137993642,
+    "C_m": 1.0,  # muF/cm^2
+    "gamma_e": 70.0e-3,  # 1/ms
+    "gamma_r": 100.0e-3,  # 1/ms
+    "g_L": 1.0,  # AU
+    "g_GABA": 1.0,  # ms
+    "g_AMPA": 1.0,  # ms
+    "g_LK": 0.018,  # mS/cm^2
+    "g_T": 3.0,  # mS/cm^2
+    "g_h": 0.062,  # mS/cm^2
+    "E_AMPA": 0.0,  # mV
+    "E_GABA": -70.0,  # mV
+    "E_L": -70.0,  # mV
+    "E_K": -100.0,  # mV
+    "E_Ca": 120.0,  # mV
+    "E_h": -40.0,  # mV
+    "alpha_Ca": -51.8e-6,  # nmol
+    "tau_Ca": 10.0,  # ms
+    "Ca_0": 2.4e-4,
+    "k1": 2.5e7,
+    "k2": 4.0e-4,
+    "k3": 1.0e-1,
+    "k4": 1.0e-3,
+    "n_P": 4.0,
+    "g_inc": 2.0,
+    "ext_current": 0.0,
+    "lambda": LAMBDA_SPEED,
+}
+DEFAULT_PARAMS_TRN = {
+    "tau": 20.0,  # ms
+    "Q_max": 400.0e-3,  # 1/ms
+    "theta": -58.5,  # mV
+    "sigma": 6.0,
+    "C1": 1.8137993642,
+    "C_m": 1.0,  # muF/cm^2
+    "gamma_e": 70.0e-3,  # 1/ms
+    "gamma_r": 100.0e-3,  # 1/ms
+    "g_L": 1.0,  # AU
+    "g_GABA": 1.0,  # ms
+    "g_AMPA": 1.0,  # ms
+    "g_LK": 0.018,  # mS/cm^2
+    "g_T": 2.3,  # mS/cm^2
+    "E_AMPA": 0.0,  # mV
+    "E_GABA": -70.0,  # mV
+    "E_L": -70.0,  # mV
+    "E_K": -100.0,  # mV
+    "E_Ca": 120.0,  # mV
+    "ext_current": 0.0,
+    "lambda": LAMBDA_SPEED,
+}
+# matrix as [to, from], masses as (TCR, TRN)
+DEFAULT_THALAMIC_CONNECTIVITY = np.array([[0.0, 5.0], [3.0, 25.0]])
+
+
+class ThalamicMass(NeuralMass):
+    """
+    Base for thalamic neural populations due to Costa et al.
+    """
+
+    name = "Thalamic mass"
+    label = "THLM"
+
+    def _get_firing_rate(self, voltage):
+        return self.params["Q_max"] / (
+            1.0 + exp(-self.params["C1"] * (voltage - self.params["theta"]) / self.params["sigma"])
+        )
+
+    # synaptic currents
+    def _get_excitatory_current(self, voltage, synaptic_rate):
+        return self.params["g_AMPA"] * synaptic_rate * (voltage - self.params["E_AMPA"])
+
+    def _get_inhibitory_current(self, voltage, synaptic_rate):
+        return self.params["g_GABA"] * synaptic_rate * (voltage - self.params["E_GABA"])
+
+    # intrinsic currents
+    def _get_leak_current(self, voltage):
+        return self.params["g_L"] * (voltage - self.params["E_L"])
+
+    def _get_potassium_leak_current(self, voltage):
+        return self.params["g_LK"] * (voltage - self.params["E_K"])
+
+    def _get_T_type_current(self, voltage, h_T_value):
+        return (
+            self.params["g_T"]
+            * self._m_inf_T(voltage)
+            * self._m_inf_T(voltage)
+            * h_T_value
+            * (voltage - self.params["E_Ca"])
+        )
+
+
+class ThalamocorticalPopulation(ThalamicMass):
```

Comment:
```suggestion
class ThalamocorticalMass(ThalamicMass):
```
I suggest, we use either "Mass" or "Population" all the way (and avoid using the other). We've said maybe "Population" is nicer and more general, but I think both is good! You decide?

Reply:
I'd go with `Mass` then :)